RecoEcal/EgammaCoreTools/interface/Mustache.h

```diff
@@ -5,18 +5,30 @@
 #include "DataFormats/CaloRecHit/interface/CaloCluster.h"
 #include "DataFormats/CaloRecHit/interface/CaloClusterFwd.h"
 #include "DataFormats/EgammaReco/interface/SuperCluster.h"
+#include "RecoEcal/EgammaCoreTools/interface/MustacheSCParametersHelper.h"
+#include "RecoEcal/EgammaCoreTools/interface/SCDynamicDPhiParametersHelper.h"
 
 namespace reco {
   namespace MustacheKernel {
-    bool inMustache(
-        const float maxEta, const float maxPhi, const float ClustE, const float ClusEta, const float ClusPhi);
-    bool inDynamicDPhiWindow(
-        const float seedEta, const float seedPhi, const float ClustE, const float ClusEta, const float clusPhi);
+    bool inMustache(const std::shared_ptr<MustacheSCParametersHelper>& params,
```

Comment:
could you pass `const MustacheSCParametersHelper*` instead, to avoid making a copy of the object in `updateSCParams`?

Reply:
Yes, I think that is possible.
You mean getting a pointer to pass with
`mustacheSCParamsHelper_ = static_cast<const reco::MustacheSCParametersHelper*>(ecalMustacheSCParamsHandle_.product());`?